flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/parser/SqlClientSyntaxHighlighter.java

```diff
@@ -0,0 +1,257 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.client.cli.parser;
+
+import org.apache.flink.table.api.SqlDialect;
+import org.apache.flink.table.api.config.TableConfigOptions;
+import org.apache.flink.table.client.config.SqlClientOptions;
+import org.apache.flink.table.client.gateway.Executor;
+
+import org.jline.reader.LineReader;
+import org.jline.reader.impl.DefaultHighlighter;
+import org.jline.utils.AttributedString;
+import org.jline.utils.AttributedStringBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** Sql Client syntax highlighter. */
+public class SqlClientSyntaxHighlighter extends DefaultHighlighter {
+    private static final Logger LOG = LoggerFactory.getLogger(SqlClientSyntaxHighlighter.class);
+    private static Set<String> flinkKeywordSet;
+    private static Set<Character> flinkKeywordCharacterSet;
+
+    static {
+        try (InputStream is =
+                SqlClientSyntaxHighlighter.class.getResourceAsStream("/keywords.properties")) {
+            Properties props = new Properties();
+            props.load(is);
+            flinkKeywordSet =
+                    Collections.unmodifiableSet(
+                            Arrays.stream(props.get("default").toString().split(";"))
+                                    .collect(Collectors.toSet()));
+            flinkKeywordCharacterSet =
+                    flinkKeywordSet.stream()
+                            .flatMap(t -> t.chars().mapToObj(c -> (char) c))
+                            .collect(Collectors.toSet());
+        } catch (IOException e) {
+            LOG.error("Exception: ", e);
+            flinkKeywordSet = Collections.emptySet();
+        }
+    }
+
+    private final Executor executor;
+
+    public SqlClientSyntaxHighlighter(Executor executor) {
+        this.executor = executor;
+    }
+
+    @Override
+    public AttributedString highlight(LineReader reader, String buffer) {
+        final SyntaxHighlightStyle.BuiltInStyle style =
+                SyntaxHighlightStyle.BuiltInStyle.fromString(
+                        executor.getSessionConfig()
+                                .get(SqlClientOptions.DISPLAY_DEFAULT_COLOR_SCHEMA));
```

Comment:
yes, it's handled at org.apache.flink.table.client.cli.parser.SyntaxHighlightStyle.BuiltInStyle#fromString.
In case of wrong schema name the default one (no highlight) will be used
https://github.com/apache/flink/pull/22063/files#diff-2be1e22afaf63582ff481966e09b1c43fc35a6176faee2f495b97f9b54562c8aR86-R95